### PR TITLE
Disable delete-protection for cognito user pool

### PR DESCRIPTION
**Pull request by suleman-sohail**
On nuking aws account, having cognito user pool, as one of its resources, nuke fails to cleanup. `deletionProtection` support is added in remove method of cognito-user-pool.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ type DisableDeletionProtection struct {
 	CloudformationStack bool `yaml:"CloudformationStack"`
 	ELBv2               bool `yaml:"ELBv2"`
 	QLDBLedger          bool `yaml:"QLDBLedger"`
+	CognitoUserPool 	bool `yaml:"CognitoUserPool"`
```
Comment (farrukh-taqveem) [inline pkg/config/config.go:50]:
```suggestion
	CognitoUserPool bool `yaml:"CognitoUserPool"`
```

File: resources/cognito-userpools.go [hunk 2]
```diff
@@ -4,12 +4,15 @@ import (
...
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/rebuy-de/aws-nuke/v2/pkg/config"
 )
 
 type CognitoUserPool struct {
 	svc  *cognitoidentityprovider.CognitoIdentityProvider
 	name *string
 	id   *string
+
```
Comment (farrukh-taqveem) [inline resources/cognito-userpools.go:14]:
```suggestion
```

File: resources/cognito-userpools.go [hunk 3]
```diff
@@ -48,11 +51,49 @@ func ListCognitoUserPools(sess *session.Session) ([]Resource, error) {
...
+				UserPoolId: f.id,
+			})
+			if err != nil {
+				return err
+			}
+			
+			return nil
+		}
+	}
+
```
Comment (farrukh-taqveem) [inline resources/cognito-userpools.go:79]:
```suggestion
```

**Review by farrukh-taqveem — COMMENTED** [review]
Let's remove these un-necessary lines & spaces. 
Rest Looks good.

**Review by farrukh-taqveem — APPROVED** [review]

**Comment by npellegrin** [comment]
Hello. I am also looking forward to have this feature in aws-nuke. Does this Merge request require additional fixes I can help with ?

**Comment by npellegrin** [comment]
References #993

**Comment by suleman-sohail** [comment]
> Hello. I am also looking forward to have this feature in aws-nuke. Does this Merge request require additional fixes I can help with ?

Nope, I believe we are all set for the referenced issue, we can get this merged.

File: resources/cognito-userpools.go [hunk 4]
```diff
@@ -48,12 +50,46 @@ func ListCognitoUserPools(sess *session.Session) ([]Resource, error) {
...
+}
 
+func (f *CognitoUserPool) Remove() error {
 	_, err := f.svc.DeleteUserPool(&cognitoidentityprovider.DeleteUserPoolInput{
 		UserPoolId: f.id,
 	})
+	if err != nil {
+		if f.featureFlags.DisableDeletionProtection.CognitoUserPool{
+			err = f.DisableProtection()
+			if err!=nil{
```
Comment (bjoernhaeuser) [inline resources/cognito-userpools.go:64]: Can you please quickly run `go fmt` on these files?

**Review by der-eismann — CHANGES_REQUESTED** [review]

**Comment by BrutalSimplicity** [comment]
Ran into this issue today. It looks like this is pending some updates from the author @suleman-sohail. If they're unable to get to this, should a new PR be opened? It doesn't look like much is left here but some cosmetic changes.

**Comment by suleman-sohail** [comment]
> Ran into this issue today. It looks like this is pending some updates from the author @suleman-sohail. If they're unable to get to this, should a new PR be opened? It doesn't look like much is left here but some cosmetic changes.

Hi, I could not reply anytime soon, thanks for waiting. I believe I was able to address the relevant comments, I might have miss something. You can create PR on child branch of this one, do any changes you think are needed for this PR to get merged, get that merged in this one, and we are good to go.

**Review by malc-moo — APPROVED** [review]

**Review by malc-moo — APPROVED** [review]

**Comment by ekristen** [comment]
This was implemented in the fork via this PR https://github.com/ekristen/aws-nuke/pull/311 and is in 3.23.0 and forward.

If you have a chance, please check it out and let us know if you run into an issues by opening an issue over on the fork.

----

Please see the copy of the notice from the README about the deprecation of this project. Sven was kind enough to grant me access to help triage and close issues and pull requests that have already been addressed in the actively maintained fork. Some additional information is located in the [welcome issue](https://github.com/ekristen/aws-nuke/issues/276) for more information. 

> [!CAUTION]
> This repository for aws-nuke is no longer being actively maintained. We recommend users to switch to the actively maintained fork of this project at [ekristen/aws-nuke](https://github.com/ekristen/aws-nuke).
> We appreciate all the support and contributions we've received throughout the life of this project. We believe that the fork will continue to provide the functionality and support that you have come to expect from aws-nuke.
> Please note that this deprecation means we will not be addressing issues, accepting pull requests, or making future releases from this repository.
> Thank you for your understanding and support.

**Pull request closed without merging**